Mage/src/main/java/mage/game/mulligan/LondonMulligan.java

```diff
@@ -93,27 +92,20 @@ public void mulligan(Game game, UUID playerId) {
         openingHandSizes.put(playerId, openingHandSizes.get(playerId) - deduction);
         int newHandSize = openingHandSizes.get(player.getId());
         if (deduction == 0) {
-            game.fireInformEvent(new StringBuilder(player.getLogName())
-                    .append(" mulligans for free.")
-                    .toString());
+            game.fireInformEvent(player.getLogName() +
+                    " mulligans for free.");
         } else {
-            game.fireInformEvent(new StringBuilder(player.getLogName())
-                    .append(" mulligans")
-                    .append(" down to ")
-                    .append(newHandSize)
-                    .append(newHandSize == 1 ? " card" : " cards").toString());
+            game.fireInformEvent(player.getLogName() +
+                    " mulligans down to " +
+                    newHandSize +
+                    (newHandSize == 1 ? " card" : " cards"));
         }
         player.drawCards(numCards, null, game);
 
-        if (player.getHand().size() > newHandSize) {
-            int cardsToDiscard = player.getHand().size() - newHandSize;
-            Cards cards = new CardsImpl();
-            cards.addAll(player.getHand());
-            TargetCard target = new TargetCard(cardsToDiscard, cardsToDiscard, Zone.HAND,
-                    new FilterCard("card" + (cardsToDiscard > 1 ? "s" : "") + " to PUT on the BOTTOM of your library (Discard for Mulligan)"));
```

Comment:
Use same text and info about mulligan -- it's not normal effect for user, so he must see mulligan hint.

Reply:
I always felt the old Mulligan prompt was awkwardly phrased. I believe this new wording makes it clearer on what you are expected to do.